src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -390,18 +398,23 @@ public RepairRun addRepairRun(Builder repairRun, Collection<RepairSegment.Builde
               segment.getFailCount()));
       } else {
 
-        repairRunBatch.add(
-            insertRepairSegmentPrepStmt.bind(
-              segment.getRunId(),
-              segment.getId(),
-              segment.getRepairUnitId(),
-              segment.getStartToken(),
-              segment.getEndToken(),
-              segment.getState().ordinal(),
-              segment.getFailCount()));
+        try {
+          repairRunBatch.add(
+              insertRepairSegmentPrepStmt.bind(
+                  segment.getRunId(),
+                  segment.getId(),
+                  segment.getRepairUnitId(),
+                  segment.getStartToken(),
+                  segment.getEndToken(),
+                  segment.getState().ordinal(),
+                  segment.getFailCount(),
+                  objectMapper.writeValueAsString(segment.getTokenRange().getTokenRanges())));
+        } catch (JsonProcessingException e) {
+          throw new ReaperException(e);
```

Comment:
i don't think a JsonProcessingException here is a valid cause to throw a ReaperException.
this can only be an IllegalStateException, since the json was put there by us in the first place.

Reply:
ok, changed to IllegalStateException